### ISSUE TITLE
[Bug]: Links in MDX should have an underline styling by default

### ISSUE THREAD

**Issue by Sidnioulz** [storybookjs/storybook#33017]
### Describe the bug

See https://www.chromatic.com/test?appId=635781f3500dd2c49e189caf&id=6913769e99ba81e2674e55f8

https://dequeuniversity.com/rules/axe/4.10/link-in-text-block?application=axeAPI

### Reproduction link

https://635781f3500dd2c49e189caf-siuwlkquds.chromatic.com/?path=/story/components-typography-documentwrapper--with-dom

### Reproduction steps

Go to repro link

### System

```bash
N/A
```

**Comment by Sidnioulz** [comment]
@MichaelArestad in case you wanna chime in on styling here

**Comment by shokuyansh** [comment]
@dosu can i work on this?

**Comment by Sidnioulz** [comment]
@shokuyansh you can!

**Comment by NikhilChowdhury27** [comment]
@Sidnioulz  Can you please review my pr
